yb-voyager/src/monitor/monitor_target_health.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package monitor
+
+import (
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/samber/lo"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/dbzm"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+const (
+	MONITOR_HEALTH_FREQUENCY_SECONDS = 15
+	METRIC_FREE_DISK_SPACE           = "free_disk_space"
+	METRIC_TOTAL_DISK_SPACE          = "total_disk_space"
+	FREE_DISK_SPACE_THREASHOLD       = 10
+	REPLICATION_GUARDRAIL_ALERT_MSG  = "Replication is enabled on the cluster, enable it only once the import-data is done ingesting data to use the disk only for the data import operation and keep the cluster stable."
+)
+
+type TargetDBForMonitorHealth interface {
+	GetYBServers() (bool, []*tgtdb.TargetConf, error)
+	RemoveConnectionsForHosts([]string) error
+	NumOfLogicalReplicationSlots() (int64, error)
+	GetClusterMetrics() (map[string]tgtdb.NodeMetrics, error) // node_uuid:metric_name:metric_value
+}
+
+// bool variable is for indicating if a node is down or up
+var nodeStatuses map[string]bool
+
+func MonitorTargetYBHealth(yb TargetDBForMonitorHealth, skipDiskUsageHealthChecks utils.BoolStr, skipReplicationChecks utils.BoolStr, skipNodeHealthChecks utils.BoolStr, displayAlertFunc func(alertMsg string)) error {
+	nodeStatuses = make(map[string]bool)
+
+	var err error
+	loadBalancerEnabled, servers, err := yb.GetYBServers()
+	if err != nil {
+		return fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	for _, server := range servers {
+		nodeStatuses[server.Host] = true
+	}
+
+	var nodeAlert string
+	for {
+		nodeAlert, err = monitorNodesStatusAndAdapt(yb, loadBalancerEnabled, skipNodeHealthChecks)
+		if err != nil {
+			log.Errorf("error monitoring the node status and adapt: %v", err)
+		}
+
+		//skipping the disk usage monitoring for now, util the metrics function changes are made
+		err = monitorDiskUsageAndAbort(yb, true)
+		if err != nil {
+			log.Errorf("error monitoring the disk and memory: %v", err)
+		}
+
+		err = monitorReplicationOnTarget(yb, skipReplicationChecks)
+		if err != nil {
+			log.Errorf("error monitoring the replication: %v", err)
+		}
+
+		if nodeAlert != "" {
+			alertMsg := color.RedString(fmt.Sprintf("Alert!\n%s\n", nodeAlert))
```

Comment:
Let's have `ALERT: xyz` (ALERT in caps, no new line). 
I feel like that will be more concise, and similar to a "log" line